mktrayicon.c

```diff
@@ -214,28 +211,57 @@ static GtkStatusIcon *create_tray_icon(char *start_icon)
 int main(int argc, char **argv)
 {
 	char *start_icon = "none";
+	char *tooltip = NULL;
+	char *pipe = NULL;
 	FILE *fifo;
 	GThread *reader;
 
 	XInitThreads(); /* see http://stackoverflow.com/a/18690540/472927 */
 	gtk_init(&argc, &argv);
 
-	if (argc == 4 && strcmp(argv[1], "-i") == 0) {
-		start_icon = argv[2];
-	}
-
 	if (argc == 1) {
-		printf("Usage: %s [-i ICON] FIFO\n", *argv);
-		printf("Listen to FIFO for system tray icon specifications\n");
+		printf("Usage: %s [-i ICON] [-t TOOLTIP] [FIFO]\n", *argv);
+		printf("Create a system tray icon as specified\n");
 		printf("\n");
 		printf("  -i ICON\tUse the specified ICON when initializing\n");
+		printf("  -t TOOLTIP\tUse the specified TOOLTIP when initializing\n");
+		printf("  if a FIFO is provided, it should be the last argument\n");
+		printf("  if a FIFO is not provided, mktrayicon will run until killed\n");
 		printf("\n");
 		printf("Report bugs at https://github.com/jonhoo/mktrayicon\n");
 		return 0;
 	}
 
+	int parsed_args = 1;
+	int c;
+	while ((c = getopt (argc, argv, "i:t:")) != -1)
+ 		switch (c)
+		{
+			case 'i':
+				start_icon = optarg;
+				parsed_args += 2;
+				break;
+			case 't':
+				tooltip = optarg;
+				parsed_args += 2;
+				break;
+			case '?':
+				fprintf(stderr, "Unknown option: %c\n", optopt);
+				return 1;
+		}
+
+
 	icon = create_tray_icon(start_icon);
-	reader = g_thread_new("watch_fifo", watch_fifo, argv[argc-1]);
+
+	if (tooltip) {
+		gtk_status_icon_set_tooltip_text(icon, tooltip);
+	}
+
+	if (parsed_args < argc) {
```

Comment:
I'm surprised `getopt` doesn't have support for positional arguments?

Reply:
It does, but is less than intuitive:
https://www.gnu.org/software/libc/manual/html_node/Example-of-Getopt.html#Example-of-Getopt

I edited the code to use this method. Now the pipe does not have to be the last argument.